networking-calico/networking_calico/plugins/calico/plugin.py

```diff
@@ -152,3 +177,19 @@ def create_floatingip(
             context.fip_update_port_id = new_floatingip["port_id"]
             self.mechanism_manager._call_on_drivers("update_floatingip", context)
         return new_floatingip
+
+
+class CalicoNotifier(object):
+    def __init__(self, plugin):
+        self.plugin = plugin
+
+    def security_groups_rule_updated(self, context, sgids):
+        self.plugin.mechanism_manager._call_on_drivers(
+            "security_groups_rule_updated", SGRUpdateContext(context, sgids)
+        )
+
+    def __getattr__(self, name):
+        def fn(*args, **kwargs):
+            LOG.info("CalicoNotifier no-op: %s %r %r", name, args, kwargs)
```

Comment:
Info might be a bit spammy?  Think this might get callbacks for lots of operations.